### PR TITLE
test(button-toggle): refactor for cypress tests

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour
Refactor for Button-Toggle and Button-Toggle-Group components to use new cypress-component-react framework for testing.
I have added test.js and deleted steps.js and feature file.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x]  Run npx cypress open-ct to check if there is newly added test.file
- [x]  Check if the button-toggle.test.js file passed
- [x]  Check if the button-toggle-group.test.js file passed
- [x]  Compare all tests were moved/refactored from .feature file -> .test.js file
- [x] Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+        </ButtonToggle>
+        <ButtonToggle name="third" key="third" value="third">
+          Third
+        </ButtonToggle>
+      </ButtonToggleGroup>
+    </div>
+  );
+};
+
+const ButtonToggleGroupInfo = (...props) => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:104]: the same question?

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+        </ButtonToggle>
+        <ButtonToggle name="third" key="third" value="third">
+          Third
+        </ButtonToggle>
+      </ButtonToggleGroup>
+    </div>
+  );
+};
+
+const ButtonToggleGroupInfoLabel = () => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:131]: the same question?

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+        </ButtonToggle>
+        <ButtonToggle name="third" key="third" value="third">
+          Third
+        </ButtonToggle>
+      </ButtonToggleGroup>
+    </div>
+  );
+};
+
+context("Testing Button-Toggle component", () => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:158]:
```suggestion
context("Testing Button-Toggle-Group component", () => {
```

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+  describe("should render Button-Toggle component", () => {
+    it("should render Button-Toggle with data-component prop set to Cypress-Test", () => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:160]:
```suggestion
    it("should render Button-Toggle-Group with data-component prop set to Cypress-Test", () => {
```

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+      CypressMountWithProviders(
+        <ButtonToggleGroupComponent data-component={testPropValue} />
+      );
+
+      buttonToggleInput()
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:165]: could we use some new locator to do not use a lot of `parent()`?

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+
+    it("should render Button-Toggle with data-element prop set to Cypress-Test", () => {
+      CypressMountWithProviders(
+        <ButtonToggleGroupComponent data-element={testPropValue} />
+      );
+
+      buttonToggleGroup().should("have.attr", "data-element", testPropValue);
+    });
+
+    it("should render Button-Toggle with data-role prop set to Cypress-Test", () => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:181]:
```suggestion
    it("should render Button-Toggle-Group with data-role prop set to Cypress-Test", () => {
```

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+      CypressMountWithProviders(
+        <ButtonToggleGroupComponent data-role={testPropValue} />
+      );
+
+      buttonToggleGroup().should("have.attr", "data-role", testPropValue);
+    });
+
+    it("should render Button-Toggle with name prop set to Cypress-Test", () => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:189]:
```suggestion
    it("should render Button-Toggle-Group with name prop set to Cypress-Test", () => {
```

File: src/components/button-toggle/button-toggle.test.js [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+import { positionOfElement } from "../../../cypress/support/helper";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const specialCharacters = [
+  "label",
+  "mp150ú¿¡üßä",
+  "!@#$%^*()_+-=~[];:.,?{}&\"'<>",
+];
+const sizes = [
+  ["small", "32px"],
```
[the remark this answers is not in the document]
Comment (stephenogorman) [reply src/components/button-toggle/button-toggle.test.js:18]: The value being verified with the second parameter is **px, so wouldn't using a number make this more awkward?

File: src/components/button-toggle/button-toggle.test.js [hunk 9]
```diff
@@ -0,0 +1,260 @@
+import * as React from "react";
```
[the remark this answers is not in the document]
Comment (stephenogorman) [reply src/components/button-toggle/button-toggle.test.js:1]: value is now verified. 
defaultChecked is verifed in the button-toggle-group.test.js

File: src/components/button-toggle-group/button-toggle-group.test.js [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+        </ButtonToggle>
+        <ButtonToggle key="baz" value="baz">
+          Baz
+        </ButtonToggle>
+      </ButtonToggleGroup>
+    </div>
+  );
+};
+
+const ButtonToggleGroupValidation = ({ ...props }) => {
```
Comment (DlgSHi) [inline src/components/button-toggle-group/button-toggle-group.test.js:51]: why do we need this ?

File: src/components/button-toggle/button-toggle.test.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (DlgSHi) [reply src/components/button-toggle/button-toggle.test.js:18]: nope, it would not, we can assert this `${pxValue}px` expression

**Review by DlgSHi — DISMISSED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**